lib/statsd.js

```diff
@@ -40,24 +41,27 @@ var Client = function (host, port, prefix, suffix, globalize, cacheDns, mock, gl
   this.port        = options.port || 8125;
   this.prefix      = options.prefix || '';
   this.suffix      = options.suffix || '';
-  this.socket      = dgram.createSocket('udp4');
+  this.socket      = options.isChild ? options.socket : dgram.createSocket('udp4');
   this.mock        = options.mock;
-  this.globalTags  = options.globalTags || [];
+  this.globalTags  = Array.isArray(options.globalTags) ? options.globalTags : [];
   this.telegraf    = options.telegraf || false;
   this.maxBufferSize = options.maxBufferSize || 0;
   this.bufferFlushInterval = options.bufferFlushInterval || 1000;
-  this.buffer = "";
+  this.bufferHolder = options.isChild ? options.bufferHolder : { buffer: "" };
 
-  if(this.maxBufferSize > 0) {
+  if(!options.isChild && this.maxBufferSize > 0) {
     this.intervalHandle = setInterval(this.onBufferFlushInterval.bind(this), this.bufferFlushInterval);
   }
 
-  if(options.cacheDns === true){
+  if (options.isChild) {
+    this.cacheDns = options.cacheDns;
```

Comment:
Fixing this. Also note that `childClient()` method does get `host` from parent client.